README.md

```diff
@@ -190,6 +190,49 @@ Three macros are provided for adding benchmark templates.
 #define BENCHMARK_TEMPLATE2(func, arg1, arg2)
 ```
 
+### Batched processing benchmarks
+
+Sometimes you want to benchmark a batch operation and compare performance for
+different values of batch size. For example, say we have some
+`ParameterizedRoutine` and we want to compare performance as a function of
+the parameter value and not the number of function calls. This could be done
+with:
+
+```c++
+static void BM_ParameterizedRoutine(benchmark::State& state) {
+  const int param = state.range(0);
+  while (state.KeepRunningBatch(param)) {
+    ParameterizedRoutine(param);
+  }
+}
+BENCHMARK(BM_ParameterizedRoutine)->Range(1<<10, 1<<18);
+```
+
+The advantage of using `KeepRunningBatch` is that each batch is counted as
+`param` number of iterations. This means the results for different batch
+sizes are directly comparable as a function of the parameter value.
+
+### Very fast benchmarks
+
+When benchmarking very fast operations (less than 3 ns per iteration), use the
```

Comment:
One other thought. This may not be the time to resolve it since we're likely separating the range-based for addition from the first batch support, but I'd like to not forget to mention this.

I think there needs to be some clear documentation on State::iterations(). I may be mistaken, but I cannot find a place saying don't bother calling this inside of the benchmarking loop. The current implementation would return the number of actual iterations + 1 which is only slightly wrong. But, range-based for would be way off and for no apparent (to the user) reason. If checking iterations inside the benchmarking loop is something that should be supported, then I don't see how the range-based for can replace KeepRunning(). Please let me know if I'm misunderstanding this.